### PR TITLE
Initial setup for basic methods. Basic Get request is working

### PR DESCRIPTION
**Pull request by Dutesier**

### CODE REVIEW

**Comment by Dutesier** [comment]
Needs: Checking for right error code && checking for right error page.
Auto-indexing support(?)
Cache support(?)

**Review by mlanca-c — APPROVED** [review]

**Comment by mlanca-c** [comment]
Also, if you need any help with anything, I can be of assistance :mechanical_arm:

**Review by mlanca-c — APPROVED** [review]

File: srcs/http/methods.cpp [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+
+    // Create a basic HTTPResponse to be populated during the method
+    resp = smt::make_shared<webserv::HTTPResponse>(
+        new webserv::HTTPResponse(200, headers, body));
+
+    // Check if method is allowed
+    bool allowed = location->m_allowed_methods.find("GET") !=
+                   location->m_allowed_methods.end();
+    if (!allowed) {
+        resp->m_status = 403; // Need to check this
```
Comment (J0Santos) [inline srcs/http/methods.cpp:86]: 405

File: srcs/http/methods.cpp [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    std::map<std::string, std::string>     headers;
+    smt::shared_ptr<webserv::HTTPResponse> resp;
+    resp = smt::make_shared<webserv::HTTPResponse>(
+        new webserv::HTTPResponse(200, headers, body));
+
+    // Check if method is allowed
+    bool allowed = location->m_allowed_methods.find("DELETE") !=
+                   location->m_allowed_methods.end();
+    if (!allowed) {
+        resp->m_status = 403;
```
Comment (J0Santos) [inline srcs/http/methods.cpp:242]: 405, just writing all of them, so I wont forget

**Pull request merged**